extensions/big-data-cluster/src/bigDataCluster/dialog/bdcDashboardModel.ts

```diff
@@ -46,21 +57,28 @@ export class BdcDashboardModel {
 	}
 
 	public async refresh(): Promise<void> {
-		await Promise.all([
-			this._clusterController.getBdcStatus(true).then(response => {
-				this._bdcStatus = response.bdcStatus;
-				this._bdcStatusLastUpdated = new Date();
-				this._onDidUpdateBdcStatus.fire(this.bdcStatus);
-			}),
-			this._clusterController.getEndPoints(true).then(response => {
-				this._endpoints = response.endPoints || [];
-				fixEndpoints(this._endpoints);
-				this._endpointsLastUpdated = new Date();
-				this._onDidUpdateEndpoints.fire(this.serviceEndpoints);
-			})
-		]).catch(error => {
-			showErrorMessage(error);
-		});
+		try {
+			if (!this._clusterController) {
+				// If this succeeds without error we know we have a clusterController at this point
+				await this.promptReconnect();
```

Comment:
what happens if user closes the connect dialog, it seems we are handling that scenario here.

Reply:
The dashboard is notified that an error has occurred but right now we don't handle that in any special way so nothing happens (it sits there spinning). The user can press refresh to cause the dialog to appear again and if the correct information is entered then the dashboard will refresh with new information as expected.

I have a follow up task to make the dashboard indicate if an error occurred but that won't be a part of this PR.